File: 프로그래머스/lv2/42586. 기능개발/기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> answerList = new ArrayList<>();

        // 반복횟수
        int temp = 1;

        // pop하는 갯수
        int count = 0;

        for(int i=0; i< progresses.length;i++){
            boolean isNotFinshed = true;
            while(isNotFinshed){
                if(100-(progresses[i]+(speeds[i]*temp)) <=0){
                    count += 1;
                    isNotFinshed = false;
                }
                else if(100-(progresses[i]+(speeds[i]*temp)) >0 && count >0){
                    answerList.add(count);
                    count = 0;
                    temp+=1;
                }
                else{
                    temp+=1;
                }
            }
        }
        answerList.add(count);
        int[] answer = new int[answerList.size()];
        for(int i =0; i< answerList.size();i++){
            answer[i] = answerList.get(i);
        }
        for(int i : answer){
            System.out.println(i);
        }
        
        
        return answer;
    }
}

File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    num = 1
    count = 0
    for i in range(len(progresses)):
        while True:
            cal = 100 - progresses[i] - speeds[i] * num
            print(cal)
            if cal>0 and count != 0:
                num += 1
                answer.append(count)
                count = 0
            elif cal > 0:
                num += 1
            elif cal <= 0 and len(progresses)-1 == i:
                count += 1
                answer.append(count)
                break
            else:
                count += 1
                break
    return answer